### ISSUE TITLE
Cleanup or replace GKE-based e2e jobs

### ISSUE THREAD

**Issue by alejandrox1** [kubernetes/kubernetes#88971]
Tl;DR every single e2e job that runs on GKE is either broken or shows up intestgrid with no results because the e2e job has already been removed but testgrid config has not been updated. 

**Which jobs are failing**:

**API Machinery Jobs** (jobs have been stopped but still show up in testgrid)
* https://testgrid.k8s.io/sig-api-machinery-gce-gke#gke
* https://testgrid.k8s.io/sig-api-machinery-gce-gke#gke-flaky
* https://testgrid.k8s.io/sig-api-machinery-gce-gke#gke-slow
* https://testgrid.k8s.io/sig-api-machinery-gce-gke#gke-serial

**SIG Apps**  (jobs have been stopped but still show up in testgrid)
* https://testgrid.k8s.io/sig-apps#gke
* https://testgrid.k8s.io/sig-apps#gke-slow
* https://testgrid.k8s.io/sig-apps#gke-serial

**SIG Auth** (jobs have been stopped but still show up in testgrid)
* https://testgrid.k8s.io/sig-auth#gke
* https://testgrid.k8s.io/sig-auth#gke-slow
* https://testgrid.k8s.io/sig-auth#gke-serial

**SIG Autoscaling** (jobs have been stopped but still show up in testgrid)
* https://testgrid.k8s.io/sig-autoscaling-hpa#gci-gke-autoscaling-hpa-cm

**SIG CLI** (broken jobs)
* https://testgrid.k8s.io/sig-cli-master#soak-gke-gci
* https://testgrid.k8s.io/sig-cli-master#gke
* https://testgrid.k8s.io/sig-cli-master#gke-serial
* https://testgrid.k8s.io/sig-cli-master#skew-cluster-latest-kubectl-stable1-gke
* etc etc

**SIG Instrumentation**  (broken jobs)
* https://testgrid.k8s.io/sig-instrumentation#gci-gke
* etc etc

**SIG Network**  (broken jobs)
* https://testgrid.k8s.io/sig-network-gke

etc etc

Im going to stop here, hopefully everyone gets the picture: there doesn't appear to be a single GKE e2e job that works.
And these jobs have been empty or failing for a very long time.

The failure can be summarized more or less by the following logs:
```
W0309 13:58:23.740] 2020/03/09 13:58:23 process.go:153: Running: gcloud container clusters create --quiet --project=gke-up-c1-2-g1-3-up-mas --zone=us-west1-b --machine-type=n1-standard-2 --image-type=gci --num-nodes=3 --network=bootstrap-e2e --cluster-version=1.19.0-alpha.0.711+5383e01d2aba76 bootstrap-e2e
W0309 13:58:24.648] WARNING: Currently VPC-native is not the default mode during cluster creation. In the future, this will become the default mode and can be disabled using `--no-enable-ip-alias` flag. Use `--[no-]enable-ip-alias` flag to suppress this warning.
W0309 13:58:24.649] WARNING: Newly created clusters and node-pools will have node auto-upgrade enabled by default. This can be disabled using the `--no-enable-autoupgrade` flag.
W0309 13:58:24.650] WARNING: Starting with version 1.18, clusters will have shielded GKE nodes by default.
W0309 13:58:24.650] WARNING: Your Pod address range (`--cluster-ipv4-cidr`) can accommodate at most 1008 node(s). 
W0309 13:58:24.650] This will disable the autorepair feature for nodes. Please see https://cloud.google.com/kubernetes-engine/docs/node-auto-repair for more information on node autorepairs.
W0309 13:58:28.730] Creating cluster bootstrap-e2e in us-west1-b...
W0309 13:58:39.103] ...................................................done.
W0309 13:58:39.228] ERROR: (gcloud.container.clusters.create) Operation [<Operation
W0309 13:58:39.228]  clusterConditions: [<StatusCondition
W0309 13:58:39.228]  message: u'Failed to create cluster'>]
W0309 13:58:39.229]  detail: u'Failed to create cluster'
W0309 13:58:39.229]  endTime: u'2020-03-09T13:58:36.987721775Z'
W0309 13:58:39.229]  name: u'operation-1583762308656-9eaa6a08'
skipped 81 lines unfold_more
W0309 14:00:29.981]   File "/workspace/./test-infra/jenkins/../scenarios/kubernetes_e2e.py", line 111, in check_env
W0309 14:00:29.981]     subprocess.check_call(cmd, env=env)
W0309 14:00:29.982]   File "/usr/lib/python2.7/subprocess.py", line 190, in check_call
W0309 14:00:29.982]     raise CalledProcessError(retcode, cmd)
W0309 14:00:29.983] subprocess.CalledProcessError: Command '('kubetest', '--dump=/workspace/_artifacts', '--gcp-service-account=/etc/service-account/service-account.json', '--up', '--down', '--test', '--deployment=gke', '--provider=gke', '--cluster=bootstrap-e2e', '--gcp-network=bootstrap-e2e', '--check-leaked-resources', '--extract=ci/latest', '--gcp-cloud-sdk=gs://cloud-sdk-testing/ci/staging', '--gcp-node-image=gci', '--gcp-zone=us-west1-b', '--gke-environment=test', '--test_args=--ginkgo.focus=\\[sig-cli\\].*\\[Serial\\]|\\[sig-cli\\].*\\[Disruptive\\] --ginkgo.skip=\\[Flaky\\]|\\[Feature:.+\\] --minStartupPods=8', '--timeout=300m')' returned non-zero exit status 1
E0309 14:00:29.994] Command failed
```
xref: https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-gci-serial-sig-cli/1237013860586622979

Notice how the `gcloud container clusters create` `--cluster-version` argument it set to 
```
--cluster-version=1.19.0-alpha.0.711+5383e01d2aba76
```
xref: GKE rapid release channel https://cloud.google.com/kubernetes-engine/docs/release-notes-rapid

I guess at this point in time we can only remove these jobs just to try and not waste resources (the jobs that do run seem to fail within first 5 mins so not a humongous waste of resources but :woman_shrugging: ) or try and replace the useful ones with GCE (?).

/cc @kubernetes/ci-signal 
/sig testing
Pinging for a pair of eyes on this. I am not sure if there is a way to fix these jobs while still using GKE.
/priority important-longterm

**Comment by spiffxp** [comment]
/assign
/close
I just noticed this issue.  The situation should be markedly improved since https://github.com/kubernetes/test-infra/pull/17213 merged 2 weeks ago

There are still some piecemeal GKE jobs floating around but I don't think any of them should be on CI Signal's radar
